### DISCUSSION TITLE
Buttons or Checkboxes

### DISCUSSION THREAD

**Discussion by magjo21** [neuefische/questions#465] [Web] [answered]
Hello✌️,

I would like to create a series of buttons.
When they are pressed, I want to save wich and how many buttons were clicked. this should give me later an evaluation. 
Until now I have tried the whole thing with normal buttons and a submit button.
However, I have difficulties to implement this, and wonder if it would be smarter with a form and checkboxes.

Would be great if someone here could give me an opinion on this, thanks.

>here is an example of what i have in mind.
![image](https://user-images.githubusercontent.com/117904136/214523912-ad385aa1-f57e-4704-8a6f-131d85338389.png)

**Comment by doemser** [comment, marked as answer]
Take a look at this prototype:

https://codesandbox.io/s/charming-danilo-du1m8q?file=/src/App.js

The bottom line is that both roads lead you to Rome with the same dataset and handler function. 

However, if you want to use a form and formData for it, it will be difficult to use the button variant without controlling the buttons with a useState, because buttons don't have a "built-in" state like checkboxes have with the "checked" prop. With checkboxes you could use formData as well.

In the end, your question feels more like a cosmetic decision.
